homeassistant/components/melnor/config_flow.py

```diff
@@ -0,0 +1,58 @@
+"""Config flow for melnor."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth.models import BluetoothServiceInfoBleak
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for melnor."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._discovered_address: str
+
```

Comment:
Looks like `async_step_user` is missing?  How does the user manually add the device?

Examples in `switchbot`

Reply:
So I was torn here when I made that decision (and an older version of this code _did_ have `async_step_user` but it didn't really do anything).

These watering valves are _super dumb_. There's no mechanism for pairing, nothing special to connect, and no way to pull a mac address off the physical device itself. The device exposes a handful of read/write gatt attributes and that is _it_.

Given this, the only interaction a user will ever have with this integration is via the discovery flow. So there's nothing for a user to manually add. It occurs to me as I write this that I could probably hook up a manual trigger for discovery in the event a user inserts batteries into a new device and then immediately attempts to add the `Melnor Bluetooth` flow, but that's about all a user could do.

Thoughts?

Reply:
In switchbot we enumerate the list of discovered devices from the bluetooth integration's history https://github.com/home-assistant/core/blob/bf510fcb4c3703de14d22b61a38b406649148a79/homeassistant/components/switchbot/config_flow.py#L195 https://github.com/home-assistant/core/blob/bf510fcb4c3703de14d22b61a38b406649148a79/homeassistant/components/switchbot/config_flow.py#L148

And then let them select one in case they dismissed the discovery they could still add it later